### PR TITLE
Some janitoring over the layout crate

### PR DESCRIPTION
**Pull request by nox**
This PR moves a bunch of code around and makes some methods from one-use traits into inherent methods.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/22684)

### CODE REVIEW

File: components/script/dom/window.rs [hunk 1]
```diff
@@ -2242,3 +2241,41 @@ impl Window {
...
+        }
+
+        //TODO: report a real filename
+        let _ = self
+            .script_chan
+            .lock()
+            .unwrap()
+            .send(ConstellationControlMsg::ReportCSSError(
+                self.pipelineid,
+                "".to_owned(),
```
Comment (emilio) [inline components/script/dom/window.rs]: We have the URL here so reporting it should be trivial, mind filing an issue for that?

File: components/layout/display_list/builder.rs [hunk 2]
```diff
@@ -2360,11 +2347,11 @@ impl BlockFlow {
             flags,
         );
 
-        if establishes_containing_block_for_absolute(
-            flags,
-            self.positioning(),
-            established_reference_frame.is_some(),
-        ) {
+        if established_reference_frame.is_some() ||
```
Comment (emilio) [inline components/layout/display_list/builder.rs]: can you do instead:

```rust
let abspos_containing_block = established_reference_frame.is_some() || ....;
if abspos_containing_block {
```

File: components/layout/display_list/builder.rs [hunk 3]
```diff
@@ -2842,11 +2829,7 @@ impl InlineFlowDisplayListBuilding for InlineFlow {
         for fragment in self.fragments.fragments.iter_mut() {
             state.containing_block_clipping_and_scrolling = previous_cb_clipping_and_scrolling;
 
-            if establishes_containing_block_for_absolute(
-                StackingContextCollectionFlags::empty(),
-                fragment.style.get_box().position,
-                false,
-            ) {
+            if fragment.style.get_box().position != StylePosition::Static {
```
Comment (emilio) [inline components/layout/display_list/builder.rs]: And same here?

This looks wrong btw, an inline can also establish a reference frame.

File: components/layout/incremental.rs [hunk 4]
```diff
@@ -22,13 +22,8 @@ bitflags! {
...
 }
 
-pub trait LayoutDamageComputation {
-    fn compute_layout_damage(self) -> SpecialRestyleDamage;
-    fn reflow_entire_document(self);
-}
-
-impl<'a> LayoutDamageComputation for &'a mut dyn Flow {
-    fn compute_layout_damage(self) -> SpecialRestyleDamage {
+impl dyn Flow {
```
Comment (emilio) [inline components/layout/incremental.rs]: Oh, this is fancy.

File: components/layout/display_list/builder.rs [hunk 5]
```diff
@@ -2828,30 +2826,24 @@ impl BaseFlow {
...
-                },
-            ) => Some(default_cursor),
-            (PointerEvents::Auto, &Cursor { keyword, .. }) => Some(keyword),
-        }
+/// Gets the cursor to use given the specific ComputedValues.  `default_cursor` specifies
+/// the cursor to use if `cursor` is `auto`. Typically, this will be `PointerCursor`, but for
+/// text display items it may be `TextCursor` or `VerticalTextCursor`.
+#[inline]
+fn get_cursor(values: &ComputedValues, default_cursor: CursorKind) -> Option<CursorKind> {
+    match (
```
Comment (emilio) [inline components/layout/display_list/builder.rs]: You could also just move it to style, your call.

In any case I think this is simpler to read as follows, if you want to make that change:

```rust
let ui = values.get_inherited_ui();
if ui.pointer_events == PointerEvents::None {
    return None;
}

match ui.cursor {
    // ...
}
```

**Review by emilio — APPROVED** [review]

File: components/layout/display_list/builder.rs (hunk 3 above)
Comment (nox) [reply components/layout/display_list/builder.rs]: Want me to file a follow-up issue? This is just the same code as before right?

File: components/script/dom/window.rs (hunk 1 above)
Comment (nox) [reply components/script/dom/window.rs]: I drive-by fixed it.

File: components/layout/display_list/builder.rs (hunk 5 above)
Comment (nox) [reply components/layout/display_list/builder.rs]: Made some changes like you suggested but didn't move it to style given there is no other user.

**Comment by CYBAI** [comment]
From Travis 👀

```
Diff in /home/travis/build/servo/servo/components/script/dom/window.rs at line 1602:
     }
 
     pub fn scroll_offset_query(&self, node: &Node) -> Vector2D<f32> {
-        if let Some(scroll_offset) = self
-            .scroll_offsets
-            .borrow()
-            .get(&node.to_opaque())
-        {
+        if let Some(scroll_offset) = self.scroll_offsets.borrow().get(&node.to_opaque()) {
             return *scroll_offset;
         }
         Vector2D::new(0.0, 0.0)
Diff in /home/travis/build/servo/servo/components/script/dom/window.rs at line 1621:
         // The scroll offsets are immediatly updated since later calls
         // to topScroll and others may access the properties before
         // webrender has a chance to update the offsets.
-        self.scroll_offsets.borrow_mut().insert(
-            node.to_opaque(),
-            Vector2D::new(x_ as f32, y_ as f32),
-        );
+        self.scroll_offsets
+            .borrow_mut()
+            .insert(node.to_opaque(), Vector2D::new(x_ as f32, y_ as f32));
 
         let NodeScrollIdResponse(scroll_id) = self.layout_rpc.node_scroll_id();
 
Run `./mach fmt` to fix the formatting
```

**Review by SimonSapin — APPROVED** [review]

**Pull request merged**